Ecal/src/Ecal/CLUE.cxx

```diff
@@ -391,131 +423,149 @@ std::vector<std::vector<const ldmx::EcalHit*>> CLUE::clustering(
+                            << std::setprecision(4) << distance_2d_prev;
             ldmx_log(trace)
-                << "      New delta_: " << std::setprecision(20) << d;
-            ldmx_log(trace)
-                << "      New delta_Z: " << std::setprecision(20) << dz;
-            current_layer[i]->delta_ = d;
-            current_layer[i]->z_delta_ = dz;
-            current_layer[i]->follower_of_ = previous_layer[j]->index_;
-          } else if (previous_layer[j]->total_energy_ <
-                     current_layer[i]->total_energy_) {
+                << "      New delta Z: " << std::setprecision(4) << dz_prev;
+            current_seed->delta_ = distance_2d_prev;
+            current_seed->z_delta_ = dz_prev;
+            current_seed->follower_of_ = prev_seed->index_;
+          } else if (prev_seed->total_energy_ < current_seed->total_energy_) {
+            ldmx_log(trace) << "      Breaking on index " << prev_seed->index_
+                            << " with energy " << prev_seed->total_energy_;
             break;
           }
         }
       }
-      if (layer + depth < nbr_of_layers_) {
+      if (layer + depth < nbr_of_layers_ && layer + depth < seeds_.size()) {
+        ldmx_log(trace) << "    Looking at post-layer: " << layer + depth;
         auto& next_layer = seeds_[layer + depth];
-        for (int j = 0; j < next_layer.size(); j++) {
-          auto d = floatDist(current_layer[i]->x_, current_layer[i]->y_,
-                             next_layer[j]->x_, next_layer[j]->y_);
-          auto dz = std::abs(current_layer[i]->z_ - next_layer[i]->z_);
-          ldmx_log(trace) << "      Delta to index_" << next_layer[j]->index_
-                          << ": " << std::setprecision(20) << d;
-          ldmx_log(trace) << "      DeltaZ to index_" << next_layer[j]->index_
-                          << ": " << std::setprecision(20) << dz;
-          if (next_layer[j]->total_energy_ > current_layer[i]->total_energy_ &&
-              d < current_layer[i]->delta_ && dz < current_layer[i]->z_delta_) {
-            ldmx_log(trace)
-                << "      New parent: index_" << next_layer[j]->index_
-                << " on layer_ " << layer + depth << "; energy "
-                << next_layer[j]->total_energy_;
+        for (const auto& next_seed : next_layer) {
+          auto distance_2d_next = floatDist(current_seed->x_, current_seed->y_,
+                                            next_seed->x_, next_seed->y_);
+          auto dz_next = std::abs(current_seed->z_ - next_seed->z_);
+          ldmx_log(trace) << "      DeltaXY to index " << next_seed->index_
+                          << ": " << std::setprecision(4) << distance_2d_next;
+          ldmx_log(trace) << "      DeltaZ to index_" << next_seed->index_
+                          << ": " << std::setprecision(4) << dz_next;
+          if (next_seed->total_energy_ > current_seed->total_energy_ &&
```

Comment:
i suppose this (avoiding index lookup) is the equivalent of my fix? what i found was a simple confusion of `i` and `j`
```
         for (int j = 0; j < nextLayer.size(); j++) {
           auto d = floatDist(currentLayer[i]->x, currentLayer[i]->y,
                              nextLayer[j]->x, nextLayer[j]->y);
-          auto dz = std::abs(currentLayer[i]->z - nextLayer[i]->z);
+          auto dz = std::abs(currentLayer[i]->z - nextLayer[j]->z);
```